### PR TITLE
Fixes incubator mutations not working on base upgrades

### LINKED ISSUE #12530: Pathogenic Incubator Bug (AGAIN)
(WEB REPORT BY: riekland REMOTE: 172.93.109.202:7777)
>Revision 2c696f47b22a07086021996cb450f73e4cd8a20b 
>General description of the issue
Unupgraded pathogenic incubators will not produce mutations, at all

>What you expected to happen
delicious BEEP

>What actually happened
foul PING

>Steps to reproduce if possible
crank up the radiation and let an incubator work

### PR DESCRIPTION
**Pull request by ghost**
I fucked up when 'buffing' and fixing the incubator in my previous PR, turns out that base level upgrades returns 0, so the initial value would be times'd by 0 and thus become 0.

fixes #12530

### CODE REVIEW

File: code/modules/virus2/dishincubator.dm [hunk 1]
```diff
@@ -48,7 +48,7 @@
 			scancount += SP.rating-1
 		if(istype(SP, /obj/item/weapon/stock_parts/micro_laser))
 			lasercount += SP.rating-1
-	mutatechance = initial(mutatechance) * scancount
+	mutatechance = initial(mutatechance) * max(1, scancount)
```
Comment (Exxion) [inline code/modules/virus2/dishincubator.dm]: I'm pretty sure you meant to use `min()`

**Comment by Exxion** [comment]
Ignore me I'm tired and also an idiot

**Comment by Exxion** [comment]
But does this make the first tier of upgrade identical to the base?

File: code/modules/virus2/dishincubator.dm [hunk 2]
```diff
@@ -48,7 +48,7 @@
 			scancount += SP.rating-1
```
Comment (Intigracy) [inline code/modules/virus2/dishincubator.dm:48]: Why don't you just += the rating as-is?

Comment (Intigracy) [reply code/modules/virus2/dishincubator.dm:48]: Assuming tiers are 1 2 3 rating, you'd get 1x 2x 3x mutatechance instead of currently with 0x 1x 2x

Comment (Intigracy) [reply code/modules/virus2/dishincubator.dm:48]: Nevermind I'm retarded and you have two each. 0x 2x 4x is current, with change it's 1x 2x 4x.

Comment (Exxion) [reply code/modules/virus2/dishincubator.dm:48]: Ah yeah
Carry on then

File: code/modules/virus2/dishincubator.dm [hunk 3]
```diff
@@ -48,7 +48,7 @@
...
 		if(istype(SP, /obj/item/weapon/stock_parts/micro_laser))
 			lasercount += SP.rating-1
-	mutatechance = initial(mutatechance) * scancount
+	mutatechance = initial(mutatechance) * max(1, scancount)
 	growthrate = initial(growthrate) + lasercount
```
Comment (Intigracy) [inline code/modules/virus2/dishincubator.dm]: Though I'm guessing this one is probably also 0?

**Comment by Intigracy** [comment]
I'm gonna merge this since mutating in virology is impossible without it.

**Pull request merged**